public/app/features/folders/state/actions.test.ts

```diff
@@ -0,0 +1,63 @@
+import { getBackendSrv, setBackendSrv, BackendSrv, FetchResponse } from '@grafana/runtime';
+import { Observable, of, throwError } from 'rxjs';
+import { thunkTester } from 'test/core/thunk/thunkTester';
+import { checkFolderPermissions } from './actions';
+import { setCanViewFolderPermissions } from './reducers';
+import { backendSrv } from 'app/core/services/backend_srv';
+import { notifyApp } from 'app/core/actions';
+import { createWarningNotification } from 'app/core/copy/appNotification';
+
+describe('folder actions', () => {
+  // let originalBackendSrv: BackendSrv;
+  let fetchSpy = jest.spyOn(backendSrv, 'fetch');
+
+  afterAll(() => {
+    fetchSpy.mockRestore();
+  });
+
+  function mockFetch(resp: Observable<any>) {
+    fetchSpy.mockReturnValueOnce(resp);
+  }
+
+  const folderUid = 'abc123';
+
+  describe('checkFolderPermissions', () => {
+    it('should dispatch true when the api call is successful', async () => {
+      mockFetch(of({}));
+
+      const dispatchedActions = await thunkTester({})
+        .givenThunk(checkFolderPermissions)
+        .whenThunkIsDispatched(folderUid);
+
+      expect(dispatchedActions).toEqual([setCanViewFolderPermissions(true)]);
+    });
+
+    it('should only dispatch false when the api call fails with 403', async () => {
```

Comment:
nit: think this test title is slightly misleading because it dispatches false if it fails for any reason.

Reply:
ahhhh yeah i see that. What I meant was "when it 403s, all it should do is dispatch false".